zkey/keystore.c

```diff
@@ -4311,16 +4311,10 @@ static int _keystore_process_cryptsetup(struct keystore *keystore,
 				printf("%s\n", cmd);
 			}
 		} else {
-			/*
-			 * Use PBKDF2 as key derivation function for LUKS2
-			 * volumes. LUKS2 uses Argon2i as default, but this
-			 * might cause out-of-memory errors when multiple LUKS2
-			 * volumes are opened automatically via /etc/crypttab
-			 */
 			util_asprintf(&cmd,
 				      "cryptsetup luksFormat %s%s--type luks2 "
 				      "--master-key-file '%s' --key-size %lu "
-				      "--cipher %s --pbkdf pbkdf2 %s%s%s",
+				      "--cipher %s %s%s%s",
```

Comment:
```suggestion
				      "--cipher %s --pbkdf argon2i --pbkdf-memory 32 --pbkdf-force-iterations 4 %s%s%s",
```

Enforce modern argon2i kdf, with low memory & time requirements, as key entropy is high and protected by hardware

Reply:
Thats certainly a good idea! The man page should also be updated accordingly, maybe keeping most of the existing description, but mention argon2i kdf, with low memory & time requirements.

Do you know if there is a minimum cryptsetup version requirement for these low memory parameters?

Reply:
It seems that such options like '--pbkdf-memory' and '--pbkdf-force-iterations' have been there for quite a while, since they were introduced with cryptsetup 2.0.0 in Dec 2017, see:
https://mirrors.edge.kernel.org/pub/linux/utils/cryptsetup/v2.0/v2.0.0-ReleaseNotes
search for 'New memory-hard PBKDF'